### DISCUSSION TITLE
Reading multi-file datasets

### DISCUSSION THREAD

**Discussion by malmans2** [pydata/xarray#5253] [Q&A]
After #5231, this currently fails:
```python
xr.open_mfdataset('my/files/*.nc', concat_dim="time",
                  data_vars='minimal', coords='minimal', compat='override')
```
ValueError: `concat_dim` can only be used with combine='nested', not with combine='by_coords'

It's copied and pasted from the documentation, so the note [here](http://xarray.pydata.org/en/latest/user-guide/io.html?highlight=reading#reading-multi-file-datasets) needs to be updated.
I'm not sure what's the best approach though... Is `concat_dim` useless and can be removed, or is it better to add `combine='nested'`?

cc: @TomNicholas @dcherian

**Comment by dcherian** [comment]
`concat_dim` is useless for `combine="by_coords"`. @TomNicholas perhaps we should raise a warning for at least one major version before making it an error?

**Reply by keewis** [reply]
and we should definitely update that note box, recommending deprecated behavior (even if we revert to a warning) is bad

**Reply by TomNicholas** [reply]
@malmans2 thanks for catching that: that doc example just needs to have `combine='nested'` added to it.

@dcherian I'm not sure what the best practice is: in the current version of xarray passing `concat_dim=X` without specifying `combine='nested'` literally doesn't do anything differently compared to leaving out `concat_dim` entirely. I suppose that means that allowing it but raising a warning saying "btw this argument doesn't do anything, you probably meant to specify `combine='nested' too" wouldn't cause any problems? Or we alternatively do raise an error, but add more detail to help the user know what they probably want to pass instead?

**Reply by dcherian** [reply]
>  I suppose that means that allowing it but raising a warning saying "btw this argument doesn't do anything, you probably meant to specify `combine='nested' too" wouldn't cause any problems?

Ya I think it's better to not break any code suddenly. This warning message covers all the relevant info I think.

**Reply by TomNicholas** [reply]
Okay, I changed it to a warning in #5255
